Lib/test/test_difflib.py

```diff
@@ -466,13 +467,44 @@ def _assert_type_error(self, msg, generator, *args):
             list(generator(*args))
         self.assertEqual(msg, str(ctx.exception))
 
+class TestJunkAPIs(unittest.TestCase):
+    def test_is_line_junk_true(self):
+        for line in ['#', '  ', ' #', '# ', ' # ', '']:
+            is_junk = difflib.IS_LINE_JUNK(line)
+            self.assertEqual(is_junk, True)
+
+    def test_is_line_junk_false(self):
+        for line in ['##', ' ##', '## ', 'not a junk line', 'abc ', 'abc #', 'abc # ']:
+            is_junk = difflib.IS_LINE_JUNK(line)
+            self.assertEqual(is_junk, False)
+
+    def test_is_line_junk_REDOS(self):
+        evil_input = ('\t' * 50000) + '##'
+
+        start = datetime.datetime.now()
+        is_junk = difflib.IS_LINE_JUNK(evil_input)
+        end = datetime.datetime.now()
+        elapsed = end - start
+
+        self.assertEqual(is_junk, False)
+        self.assertEqual(elapsed.seconds, 0)
```

Comment:
Timing test is very fragile. It can fail randomly when tests are paused for some reasons. I suggest to remove the timing test and increase the size of input so that it would take a hour with non-fixed code.

Reply:
Then the test will pass if you wait long enough. Since the test suite already takes a long time to run and does so in parallel (so one hanging test might not be visible), this seems undesirable to me. Thoughts?

Reply:
Same thing in Lib/test/test_poplib.py

Reply:
If the test takes too long time for running it will be failed on buildbots due to total timeout.